src/components/AddPaymentMethodMenu.js

```diff
@@ -9,54 +9,166 @@ import styles from '../styles/styles';
-        />
-        <MenuItem
-            title={props.translate('common.debitCard')}
-            icon={Expensicons.CreditCard}
-            onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.DEBIT_CARD)}
-            wrapperStyle={styles.pr15}
-        />
-        {!props.payPalMeUsername && (
-            <MenuItem
-                title={props.translate('common.payPalMe')}
-                icon={Expensicons.PayPal}
-                onPress={() => props.onItemSelected(CONST.PAYMENT_METHODS.PAYPAL)}
-                wrapperStyle={styles.pr15}
-            />
-        )}
-    </Popover>
-);
+class AddPaymentMethodMenu extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isAddPaymentMenuActive: false,
+            anchorPosition: {
+                top: 0,
+                left: 0,
+            },
+        };
+        this.addPaymentMethodPressed = this.addPaymentMethodPressed.bind(this);
+        this.addPaymentMethodTypePressed = this.addPaymentMethodTypePressed.bind(this);
+        this.hideAddPaymentMenu = this.hideAddPaymentMenu.bind(this);
+    }
+
+    /**
+     * Get the AddPaymentMethod Button title
+     * @returns {String}
+     */
+    getAddPaymentMethodButtonTitle() {
+        switch (this.props.filterType) {
+            case CONST.PAYMENT_METHODS.BANK_ACCOUNT:
+                return this.props.translate('paymentMethodList.addBankAccount');
+            case CONST.PAYMENT_METHODS.DEBIT_CARD:
+                return this.props.translate('paymentMethodList.addDebitCard');
+            default: break;
+        }
+        return this.props.translate('paymentMethodList.addPaymentMethod');
+    }
+
+    /**
+     * Display the add payment method menu
+     * @param {Object} nativeEvent
+     */
+    addPaymentMethodPressed(nativeEvent) {
+        const position = getClickedElementLocation(nativeEvent);
+        this.setState({
+            isAddPaymentMenuActive: true,
+            anchorPosition: {
+                top: position.bottom,
+
+                // We want the position to be 20px to the right of the left border
+                left: position.left + 20,
+            },
+        });
+    }
+
+    /**
+     * Navigate to the appropriate payment type addition screen
+     * @param {String} paymentType
+     */
+    addPaymentMethodTypePressed(paymentType) {
+        if (!this.props.filterType) {
+            this.hideAddPaymentMenu();
+        }
+
+        if (paymentType === CONST.PAYMENT_METHODS.PAYPAL) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_PAYPAL_ME);
+            return;
+        }
+
+        if (paymentType === CONST.PAYMENT_METHODS.DEBIT_CARD) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_DEBIT_CARD);
+            return;
+        }
+
+        if (paymentType === CONST.PAYMENT_METHODS.BANK_ACCOUNT) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_BANK_ACCOUNT);
+            return;
+        }
```

Comment:
None of these routes should be in this component because we will eventually need to navigate to different routes depending on the context in which this component will be used. I would suggest passing in the route as a prop to keep this component reusable.